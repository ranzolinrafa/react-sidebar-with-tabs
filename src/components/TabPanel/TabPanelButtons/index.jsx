import { memo, useMemo } from 'react';
import { filterTabs } from '../../../configs/tabs';
import * as S from './styles';

const TabPanelButtons = ({ setTabActive, onClick, tabs }) => {
    const filteredTabs = useMemo(() => filterTabs(tabs), [tabs]);

    const handleClick = (tab) => {
        onClick();
        setTabActive(tab);
    };

    return (
        <S.ButtonGroup>
            {filteredTabs.map((item) => {
                const IconComponent = item.icon;
                return (
                    <button
                        key={item.name}
                        onClick={() => handleClick(item.name)}
                    >
                        <IconComponent />
                    </button>
                );
            })}
        </S.ButtonGroup>
    );
};

export default memo(TabPanelButtons);
