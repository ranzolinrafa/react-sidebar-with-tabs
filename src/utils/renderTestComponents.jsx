import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { dark } from '../styles/theme';

export const renderWithTheme = (children) => {
    return render(<ThemeProvider theme={dark}>{children}</ThemeProvider>);
};
