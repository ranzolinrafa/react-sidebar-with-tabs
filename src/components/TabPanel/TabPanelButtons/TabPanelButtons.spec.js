import { renderWithTheme } from '../../../utils/renderTestComponents';
import TabPanelButtons from '.';
import tabsConfig from '../../../configs/tabs';

describe('<TabPanelButtons />', () => {
    it('should render based in a config object', () => {
        let tabs = [];
        tabsConfig.forEach((item) => tabs.push(item.name));

        const { container } = renderWithTheme(<TabPanelButtons tabs={tabs} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
