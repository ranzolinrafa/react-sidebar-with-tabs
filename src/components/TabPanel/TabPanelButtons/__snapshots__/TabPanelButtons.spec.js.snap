// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabPanelButtons /> should render based in a config object 1`] = `
<div
  class="sc-furwcr kDnHPI"
>
  <button>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 2a2 2 0 0 0-2-2H2a2 2 0 0 0-2 2v8a2 2 0 0 0 2 2h9.586a1 1 0 0 1 .707.293l2.853 2.853a.5.5 0 0 0 .854-.353V2zM3.5 3h9a.5.5 0 0 1 0 1h-9a.5.5 0 0 1 0-1zm0 2.5h9a.5.5 0 0 1 0 1h-9a.5.5 0 0 1 0-1zm0 2.5h5a.5.5 0 0 1 0 1h-5a.5.5 0 0 1 0-1z"
      />
    </svg>
  </button>
  <button>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5 11.5a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9a.5.5 0 0 1-.5-.5zm0-4a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9a.5.5 0 0 1-.5-.5zm0-4a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9a.5.5 0 0 1-.5-.5zM3.854 2.146a.5.5 0 0 1 0 .708l-1.5 1.5a.5.5 0 0 1-.708 0l-.5-.5a.5.5 0 1 1 .708-.708L2 3.293l1.146-1.147a.5.5 0 0 1 .708 0zm0 4a.5.5 0 0 1 0 .708l-1.5 1.5a.5.5 0 0 1-.708 0l-.5-.5a.5.5 0 1 1 .708-.708L2 7.293l1.146-1.147a.5.5 0 0 1 .708 0zm0 4a.5.5 0 0 1 0 .708l-1.5 1.5a.5.5 0 0 1-.708 0l-.5-.5a.5.5 0 0 1 .708-.708l.146.147 1.146-1.147a.5.5 0 0 1 .708 0z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
  <button>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M112 224c61.9 0 112-50.1 112-112S173.9 0 112 0 0 50.1 0 112s50.1 112 112 112zm0-160c26.5 0 48 21.5 48 48s-21.5 48-48 48-48-21.5-48-48 21.5-48 48-48zm224 224c-61.9 0-112 50.1-112 112s50.1 112 112 112 112-50.1 112-112-50.1-112-112-112zm0 160c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zM392.3.2l31.6-.1c19.4-.1 30.9 21.8 19.7 37.8L77.4 501.6a23.95 23.95 0 0 1-19.6 10.2l-33.4.1c-19.5 0-30.9-21.9-19.7-37.8l368-463.7C377.2 4 384.5.2 392.3.2z"
      />
    </svg>
  </button>
</div>
`;
